server/src/test/java/org/apache/druid/server/coordinator/loading/HttpLoadQueuePeonTest.java

```diff
@@ -82,29 +78,26 @@ public class HttpLoadQueuePeonTest
   public void setUp()
   {
     httpClient = new TestHttpClient();
-    JacksonConfigManager configManager = EasyMock.createNiceMock(JacksonConfigManager.class);
-    EasyMock.expect(
-        configManager.watch(
-            EasyMock.eq(CoordinatorDynamicConfig.CONFIG_KEY),
-            EasyMock.anyObject(Class.class),
-            EasyMock.anyObject()
-        )
-    ).andReturn(new AtomicReference<>(CoordinatorDynamicConfig.builder().build())).anyTimes();
-    EasyMock.replay(configManager);
     httpLoadQueuePeon = new HttpLoadQueuePeon(
         "http://dummy:4000",
         MAPPER,
         httpClient,
         new HttpLoadQueuePeonConfig(null, null, 10),
+        () -> SegmentLoadingMode.NORMAL,
         new WrappingScheduledExecutorService(
             "HttpLoadQueuePeonTest-%s",
             httpClient.processingExecutor,
             true
         ),
-        httpClient.callbackExecutor,
-        () -> SegmentLoadingMode.NORMAL,
-        new HistoricalLoadingCapabilities(1, 3)
-    );
+        httpClient.callbackExecutor
+    )
+    {
+      @Override
+      SegmentLoadingCapabilities fetchSegmentLoadingCapabilities()
```

Comment:
Non-blocker:
Do we need to override this?
Can't we make the `HttpClient` handle this API to return what we want?